docs: fix internal cross-page references by injecting .md

hirenp opened the pull request:
Fixing external references by injecting .md so that links works correctly while reading the documentation.

ArangoGutierrez reviewed (changes requested):
Thanks for this PR! could you update the PR description with a small line detailing the change, so we keep a nice git history.

/ok-to-test

hirenp commented:
> Something like this should give a good starting point, doing most of the work but possibly doing some incorrect edits and missing some, too:
Thanks! That seemed to have done the trick. I tried spot checking a couple places and they look correct to me.

marquiz commented on docs/get-started/introduction.md:
```diff
@@ -43,14 +43,14 @@ NFD-Topology-Updater is a daemon responsible for examining allocated
 to new pod on a per-zone basis (where a zone can be a NUMA node). It then
 creates or updates a
-[NodeResourceTopology](../usage/custom-resources#noderesourcetopology) custom
+[NodeResourceTopology](../usage/custom-resourcesnoderesourcetopology) custom
```
```suggestion
[NodeResourceTopology](../usage/custom-resources.md#noderesourcetopology) custom
```

marquiz commented on docs/get-started/introduction.md:
```diff
@@ -43,14 +43,14 @@ NFD-Topology-Updater is a daemon responsible for examining allocated
 
 NFD-Topology-Garbage-Collector is a daemon responsible for cleaning obsolete
-[NodeResourceTopology](../usage/custom-resources#noderesourcetopology) objects,
+[NodeResourceTopology](../usage/custom-resourcesnoderesourcetopology) objects,
```
```suggestion
[NodeResourceTopology](../usage/custom-resources.md#noderesourcetopology) objects,
```

marquiz commented on docs/get-started/introduction.md:
```diff
@@ -114,13 +114,13 @@ only created on nodes running nfd-master.
 
 NFD-Topology-Updater creates
-[NodeResourceTopology](../usage/custom-resources#noderesourcetopology) objects
+[NodeResourceTopology](../usage/custom-resourcesnoderesourcetopology) objects
```
```suggestion
[NodeResourceTopology](../usage/custom-resources.md#noderesourcetopology) objects
```

marquiz commented on docs/usage/nfd-topology-gc.md:
```diff
@@ -24,6 +24,6 @@ when no -gc-interval is specified.
 
 In Helm deployments,
-(see [Topology Garbage Collector](../deployment/helm.md#topology-garbage-collector-parameters)
+(see [Topology Garbage Collector](../deployment/helmtopology-garbage-collector-parameters)
```
```suggestion
(see [Topology Garbage Collector](../deployment/helm.md#topology-garbage-collector-parameters)
```

marquiz commented on docs/usage/nfd-topology-updater.md:
```diff
@@ -12,18 +12,18 @@ sort: 5
 -sleep-interval is specified.
 In addition, it can avoid examining specific allocated resources
-given a configuration of resources to exclude via [`-excludeList`](../reference/topology-updater-configuration-reference.md#excludelist)
+given a configuration of resources to exclude via [`-excludeList`](../reference/topology-updater-configuration-referenceexcludelist)
```
```suggestion
given a configuration of resources to exclude via [`-excludeList`](../reference/topology-updater-configuration-reference.md#excludelist)
```

marquiz commented on docs/usage/nfd-topology-updater.md:
```diff
@@ -60,7 +60,7 @@ kubectl -n ${NFD_NS} edit configmap nfd-topology-updater-conf
 
 In Helm deployments,
-[Topology Updater parameters](../deployment/helm.md#topology-updater-parameters)
+[Topology Updater parameters](../deployment/helmtopology-updater-parameters)
```
```suggestion
[Topology Updater parameters](../deployment/helm.md#topology-updater-parameters)
```

marquiz commented on docs/usage/nfd-worker.md:
```diff
@@ -22,7 +22,7 @@ Worker connects to the nfd-master service to advertise hardware features.
 When run as a daemonset, nodes are re-labeled at an default interval of 60s.
 This can be changed by using the
-[`core.sleepInterval`](../reference/worker-configuration-reference.html#coresleepinterval)
+[`core.sleepInterval`](../reference/worker-configuration-reference.html.md#coresleepinterval)
```
```suggestion
[`core.sleepInterval`](../reference/worker-configuration-reference.md#coresleepinterval)
```

marquiz reviewed (commented):
Thanks for the updater @hirenp. I found some mistakes, possibly because of erroneous sed script that I provided if you used them.

Could you also squash the PR into one commit and adjust the commit message accordingly, e.g. `"docs: fix internal cross-page references by injecting .md"` or smth

hirenp commented:
> Thanks for the updater @hirenp. I found some mistakes, possibly because of erroneous sed script that I provided if you used them.
> 
> Could you also squash the PR into one commit and adjust the commit message accordingly, e.g. `"docs: fix internal cross-page references by injecting .md"` or smth

Apologies for missing those. I fixed as per your suggestions.

marquiz commented:
> Apologies for missing those. I fixed as per your suggestions.

Np, thanks for the update @hirenp. Looks good, now. Just one last nit: could you remove the inadvertent quotation mark (`"`, a copy-paste mistake(?))  from the end of the commit message

/retitle docs: fix internal cross-page references by injecting .md

hirenp commented:
> > Apologies for missing those. I fixed as per your suggestions.
> 
> Np, thanks for the update @hirenp. Looks good, now. Just one last nit: could you remove the inadvertent quotation mark (`"`, a copy-paste mistake(?)) from the end of the commit message

:man_facepalming: 

Fixed. Thanks.

marquiz reviewed (approved):
👍

fmuyassarov reviewed (commented):
/lgtm
